src/components/project_admin/ag_suites/ag_suites.vue

```diff
@@ -0,0 +1,608 @@
+  update_ag_test_command_changed(ag_test_command: AGTestCommand): void {
+    let parent_suite: AGTestSuite = this.active_suite!;
+    let parent_case: AGTestCase = this.active_case!;
+
+    for (let suite_index = 0; suite_index < this.test_suites.length; ++suite_index) {
+      let case_index = this.test_suites[suite_index].ag_test_cases.findIndex(
+        (ag_case: AGTestCase) => ag_case.pk === ag_test_command.ag_test_case);
+      if (case_index !== -1) {
+        parent_suite = this.test_suites[suite_index];
+        parent_case = parent_suite.ag_test_cases[case_index];
+      }
+    }
+
+    let command_index = parent_case.ag_test_commands.findIndex(
+      (ag_command) => ag_command.pk === ag_test_command.pk);
+
+    Vue.set(
+      parent_case.ag_test_commands,
+      command_index,
+      ag_test_command
+    );
+  }
+
+  update_ag_test_command_created(ag_test_command: AGTestCommand): void {
+    let parent_suite: AGTestSuite = this.active_suite!;
+    let parent_case: AGTestCase = this.active_case!;
+
+    for (let suite_index = 0; suite_index < this.test_suites.length; ++suite_index) {
+      let case_index = this.test_suites[suite_index].ag_test_cases.findIndex(
+        (ag_case: AGTestCase) => ag_case.pk === ag_test_command.ag_test_case);
+      if (case_index !== -1) {
+        parent_suite = this.test_suites[suite_index];
+        parent_case = parent_suite.ag_test_cases[case_index];
+      }
+    }
+
+    parent_case!.ag_test_commands.push(ag_test_command);
+    this.update_active_command(ag_test_command,
+                               parent_case,
+                               parent_suite);
+  }
+
+  update_ag_test_command_deleted(ag_test_command: AGTestCommand): void {
+    let parent_suite: AGTestSuite = this.active_suite!;
+    let parent_case: AGTestCase = this.active_case!;
+
+    for (let suite_index = 0; suite_index < this.test_suites.length; ++suite_index) {
+      let case_index = this.test_suites[suite_index].ag_test_cases.findIndex(
+        (ag_case: AGTestCase) => ag_case.pk === ag_test_command.ag_test_case);
+      if (case_index !== -1) {
+        parent_suite = this.test_suites[suite_index];
+        parent_case = parent_suite.ag_test_cases[case_index];
+      }
+    }
+
+    let command_index = parent_case.ag_test_commands.findIndex(
+      (ag_command) => ag_command.pk === ag_test_command.pk);
+
+    let command_was_active = this.active_command !== null
+                             && ag_test_command.pk === this.active_command!.pk;
+    this.reset_active_indices();
+    parent_case!.ag_test_commands.splice(command_index, 1);
+    if (command_was_active) {
+      if (command_index === parent_case!.ag_test_commands.length) {
+        this.update_active_command(parent_case!.ag_test_commands[command_index - 1],
+                                   parent_case,
+                                   parent_suite);
+      }
+      else {
```

Comment:
Does it work without this else (and removing the earlier call to reset_active_indices)?